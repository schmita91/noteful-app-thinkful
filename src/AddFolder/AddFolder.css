#folder-name {
  display: flex;
  justify-content: center;
  min-width: 300px;
}
.add-new-folder {
  min-width: 250px;
  display: flex;
  justify-content: center;
}