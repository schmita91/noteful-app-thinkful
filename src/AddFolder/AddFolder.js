import React, { Component } from 'react';
import ApiContext from '../ApiContext';
import { apiEndpoint } from '../config';
import ErrorBoundary from '../ErrorBoundary/ErrorBoundary';

export default class AddFolder extends Component {
  constructor(props) {
    super(props);
    this.state = { value: '' };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  static defaultProps = {
    history: {
      goBack: () => {},
    },
  };

  static contextType = ApiContext;

  handleChange(e) {
    this.setState({ value: e.target.value });
  }

  handleSubmit(e) {
    e.preventDefault();
    const target = e.target,
      name = target.folderName.value,
      myHeaders = new Headers();
    myHeaders.append('Content-Type', 'application/json');

    const raw = JSON.stringify({ folder_name: name });

    const requestOptions = {
      method: 'POST',
      headers: myHeaders,
      body: raw,
      redirect: 'follow',
    };

    fetch(`${apiEndpoint.API_ENDPOINT}/folders`, requestOptions)
      .then((response) => response.json())
      .then((result) => {
        this.context.addFolder(result);
        this.props.history.goBack();
      })
      .catch((error) => console.log('error', error));
  }

  render() {
    return (
      <ErrorBoundary>
        <form onSubmit={this.handleSubmit}>
          <label>
            New Folder:{' '}
            <input
              type="text"
              name="folderName"
              id="folderName"
              onChange={this.handleChange}
              placeholder="ex: Awesome folder"
              required
            />{' '}
          </label>
          <input type="submit" value="Submit" />
        </form>
      </ErrorBoundary>
    );
  }
}