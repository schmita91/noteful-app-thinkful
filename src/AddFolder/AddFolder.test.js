import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import AddFolder from './AddFolder'


const state = {
    error: null
};
const folderName = 'name';

describe('component addfolder',()=>{
it('renders to UI as expected', ()=>{
    const tree = renderer.create(
        <AddFolder state={state} folderName={folderName}/>
    ).toJSON()
    expect(tree).toMatchSnapshot();

})
})