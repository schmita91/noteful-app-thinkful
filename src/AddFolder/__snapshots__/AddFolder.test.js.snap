// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component addfolder renders to UI as expected 1`] = `
<div
  className="add-new-folder"
>
  <form
    className="add-folder-form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="folder-name"
      id="folder-name"
    >
      Enter the folder name:
    </label>
    <input
      defaultValue="Folder Name"
      id="folder-name"
      name="folder-name"
      type="text"
    />
    <button
      id="folder-submit"
      type="submit"
    >
      Submit
    </button>
    <button
      className="back"
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </form>
</div>
`;
